node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -1136,6 +1158,13 @@ async fn wait_until_next_check(last_poll: Instant) -> Instant {
 	Instant::now()
 }
 
+fn infinite_stream(every: Duration) -> impl FusedStream<Item = ()> {
```

Comment:
This is pretty much the same as `Metronome` right?

Reply:
I think we have multiple implementations of the same thing, there is also one in pvf. But refactoring all of that is out of scope of this PR. Also not sure we want to add a dependency just for this small thing.

Reply:
Fair enough :)